components/dropdown/dropdown-popover-mixin.js

```diff
@@ -0,0 +1,342 @@
+			'dropdown-no-header': !this._hasHeaderSlotContent,
+			'dropdown-no-padding': this.noPaddingHeader,
+			'dropdown-header-scroll': this._blockStartScroll
+		};
+		const footerClasses = {
+			'dropdown-footer': true,
+			'dropdown-no-footer': !(this._hasFooterSlotContent || (this._mobile && this._mobileTrayLocation)),
+			'dropdown-no-padding': this.noPaddingFooter,
+			'dropdown-footer-scroll': this._blockEndScroll
+		};
+
+		const closeButtonStyles = this.#getMobileCloseButtonStyles();
+
+		const content = html`
+			<div class="dropdown-content-layout" style="${styleMap(contentLayoutStyles)}">
+				<div class="${classMap(headerClasses)}">
+					<slot name="header" @slotchange="${this.#handleHeaderSlotChange}"></slot>
+				</div>
+				<div class="${classMap(contentClasses)}" @scroll="${this.#toggleScrollStyles}">
+					<slot></slot>
+				</div>
+				<div class="${classMap(footerClasses)}">
+					<slot name="footer" @slotchange="${this.#handleFooterSlotChange}"></slot>
+					<d2l-button style=${styleMap(closeButtonStyles)} @click=${this.close}>
+						${this.localize('components.dropdown.close')}
+					</d2l-button>
+				</div>
+			</div>			
+		`;
+
+		return this.renderPopover(content);
+	}
+
+	willUpdate(changedProperties) {
+		if (changedProperties.has('align') || changedProperties.has('maxHeight') || changedProperties.has('maxWidth') || changedProperties.has('minHeight') || changedProperties.has('minWidth') || changedProperties.has('mobileBreakpointOverride') || changedProperties.has('mobileTray') || changedProperties.has('noAutoClose') || changedProperties.has('noAutoFit') || changedProperties.has('noAutoFocus') || changedProperties.has('noPointer') || changedProperties.has('trapFocus') || changedProperties.has('verticalOffset')) {
+			super.configure({
+				maxHeight: this.maxHeight,
+				maxWidth: this.maxWidth,
+				minHeight: this.minHeight,
+				minWidth: this.minWidth,
+				mobileBreakpoint: this.mobileBreakpointOverride,
+				mobileTrayLocation: this.#adaptMobileTrayLocation(this.mobileTray),
+				noAutoClose: this.noAutoClose,
+				noAutoFit: this.noAutoFit,
+				noAutoFocus: this.noAutoFocus,
+				noPointer: this.noPointer,
+				offset: (this.verticalOffset !== undefined ? Number.parseInt(this.verticalOffset) : undefined),
+				position: { location: 'block-end', span: this.#adaptPositionSpan(this.align) },
+				trapFocus: this.trapFocus
+			});
+		}
+
+		if (changedProperties.has('opened')) {
+			if (this.opened) this.open(true);
+			else if (changedProperties.get('opened')) this.close();
+		}
+	}
+
+	#contentElement;
+
+	#adaptMobileTrayLocation(val) {
+		switch (val) {
+			case 'bottom': return 'block-end';
+			case 'left': return 'inline-start';
+			case 'right': return 'inline-end';
+			default: return undefined;
+		  }
+	}
+
+	#adaptPositionSpan(val) {
+		switch (val) {
+			case 'start': return 'end';
+			case 'end': return 'start';
+			default: return 'all';
+		  }
+	}
+
+	#getMobileCloseButtonStyles() {
```

Comment:
This code is the same as in the old implementation, except that it's tidied up a bit, and uses `marginBlock` and `marginInline` to avoid the RTL check that was done previously.